src/parser/index.ts

```diff
@@ -164,13 +164,24 @@ const removeTitle = (document: string) => {
 };
 
 /**
- * Cuts a string down to the specified length if needed
+ * Cuts a string down to the discord character limit and corrects missing codeblock backticks if needed
  * Appends '...' if the string was trimmed
+ * Appends '...```' if the codeblock backticks were corrected
  * @param {string} document
  * @param {number} length
  */
-const truncateString = (document: string, length: number) => {
-    return document.length > length ? document.slice(0, length - 3) + '...' : document;
+const truncateString = (document: string) => {
+    const MAX_LENGTH = 1024;
+    // Returns original document if its length is smaller than the discord maximum length
+    if (MAX_LENGTH > document.length) return document;
+    // Prepare document for truncate, appending ellipsis
+    const truncatedStr = document.slice(0, MAX_LENGTH - 3) + '...';
+    // Splits truncated string on \n and \r characters 
+    let lines = truncatedStr.match(/[^\r\n]+/g);
+    // Filter lines for codeblock backticks, store array length
+    let backtickMatches = lines?.filter(item => item === '```' || item === '```js').length;
+    // If the number of backtick instances is odd, returns document with an additional 3 backticks alongside the ellipsis. Else, return original truncated string
+    if (backtickMatches) return backtickMatches % 2 !== 0 ? document.slice(0, MAX_LENGTH - 6) + '...```' : truncatedStr;
```

Comment:
nit: I'd like to add a newline before the backticks. I am unsure if the newline character counts as part of the total length. But it would be nice to do `document.slice(0, MAX_LENGTH - 6) + '...\n```'`.

I think for now we could  consider this one additional char, so subtract 7 instead of 6.